src/utils/SpokeUtils.ts

```diff
@@ -89,3 +93,40 @@ export function isZeroAddress(address: string): boolean {
 export function getMessageHash(message: string): string {
   return isMessageEmpty(message) ? ZERO_BYTES : keccak256(message);
 }
+
+function _getRelayDataHashSvm(relayData: RelayData, destinationChainId: number): string {
+  const bufferFromHexString = (hex: string, littleEndian: boolean = false) => {
+    const buffer = Buffer.from(hex.slice(2), "hex");
+    if (buffer.length < 32) {
+      const zeroPad = Buffer.alloc(32);
+      buffer.copy(zeroPad, 32 - buffer.length);
+      return littleEndian ? zeroPad.reverse() : zeroPad;
+    }
+    return littleEndian ? buffer.slice(0, 32).reverse() : buffer.slice(0, 32);
+  };
+  const bufferFromInt = (num: BigNumber, byteLength: number, littleEndian: boolean = true) => {
+    const buffer = Buffer.from(num.toHexString().slice(2), "hex");
+    if (buffer.length < byteLength) {
+      const zeroPad = Buffer.alloc(byteLength);
+      buffer.copy(zeroPad, byteLength - buffer.length);
+      return littleEndian ? zeroPad.reverse() : zeroPad;
+    }
+    return littleEndian ? buffer.slice(0, byteLength).reverse() : buffer.slice(0, byteLength);
+  };
+  const contentToHash = Buffer.concat([
+    bufferFromHexString(relayData.depositor),
+    bufferFromHexString(relayData.recipient),
+    bufferFromHexString(relayData.exclusiveRelayer),
+    bufferFromHexString(relayData.inputToken),
+    bufferFromHexString(relayData.outputToken),
+    bufferFromInt(relayData.inputAmount, 8),
+    bufferFromInt(relayData.outputAmount, 8),
+    bufferFromInt(toBN(relayData.originChainId), 8),
+    bufferFromInt(relayData.depositId, 32, false),
+    bufferFromInt(toBN(relayData.fillDeadline), 4),
+    bufferFromInt(toBN(relayData.exclusivityDeadline), 4),
+    bufferFromHexString(getMessageHash(relayData.message)),
+    bufferFromInt(toBN(destinationChainId), 8),
+  ]);
+  return keccak256(contentToHash);
+}
```

Comment:
Damn; this one caught me by surprise. Can we stick it in `src/arch/svm/SpokeUtils.ts` ? We might also want to relocate the EVM part as well, so `getRelayDataHash()` should be pretty thin as a result.

Reply:
I agree. This was moved https://github.com/across-protocol/sdk/pull/980/commits/6186021dc7a654baf1f85ca1268914ed9ced8d73